src/Microsoft.VisualStudio.Threading.Analyzers/DiagnosticAnalyzerBase.cs

```diff
@@ -76,7 +76,7 @@ private static bool ConformsToAwaiterPattern(ITypeSymbol typeSymbol)
                 hasOnCompletedMethod |= member.Name == nameof(TaskAwaiter.OnCompleted) && member is IMethodSymbol;
                 hasIsCompletedProperty |= member.Name == nameof(TaskAwaiter.IsCompleted) && member is IPropertySymbol;
 
-                if (hasGetResultMethod || hasOnCompletedMethod || hasIsCompletedProperty)
+                if (hasGetResultMethod && hasOnCompletedMethod && hasIsCompletedProperty)
```

Comment:
Is there no test coverage for detecting the proper awaiter pattern?

Reply:
Not in depth, no. (evidently).